pkg/util/fielderrors/fielderrors.go

```diff
@@ -113,9 +113,13 @@ func NewFieldInvalid(field string, value interface{}, detail string) *Validation
 	return &ValidationError{ValidationErrorTypeInvalid, field, value, detail}
 }
 
-// NewFieldNotSupported returns a *ValidationError indicating "unsupported value"
-func NewFieldNotSupported(field string, value interface{}) *ValidationError {
-	return &ValidationError{ValidationErrorTypeNotSupported, field, value, ""}
+// NewFieldValueNotSupported returns a *ValidationError indicating "unsupported value"
+func NewFieldValueNotSupported(field string, value interface{}, validValues []string) *ValidationError {
+	detail := ""
+	if validValues != nil && len(validValues) > 0 {
```

Comment:
this isn't hurting anything, so you don't need to change it, but just ```if len(validValues) > 0 {``` is safe and sufficient.